### PR TITLE
Add default delta handler

### PR DESCRIPTION
**Pull request by hatched**
As Juju adds more delta types we need to safely continue on if they aren't yet supported in the GUI.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

File: app/models/models.js [hunk 1]
```diff
@@ -80,6 +80,11 @@ YUI.add('juju-models', function(Y) {
           // This must be from a LazyModelList.
           var changed = {};
           Y.each(data, function(value, key) {
+            if (value === undefined) {
+              // A delta in the real environment doesn't send undefined
+              // values so this makes the simulated environment work properly.
+              return;
+            }
```
Comment (hatched) [inline app/models/models.js]: Just a drive-by to make the data handling more consistent

File: app/models/models.js [hunk 2]
```diff
@@ -2042,20 +2047,6 @@ YUI.add('juju-models', function(Y) {
...
     },
 
-    /**
-      Returns a modelList given the model name.
-
-      @method getModelListByModelName
-      @param {String} modelName The model's name.
-      @return {Object} The model list.
-    */
-    getModelListByModelName: function(modelName) {
```
Comment (hatched) [inline app/models/models.js]: This method is no longer used by any other code in the app.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2450/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2451/
Test FAILed.

**Comment by hatched** [comment]
Spurious...

File: app/models/handlers.js [hunk 3]
```diff
@@ -135,37 +135,19 @@ YUI.add('juju-delta-handlers', function(Y) {
...
+
+      @method defaultHandler
+      @param {Object} db The application db (unused)
+      @param {String} action The action which the delta is trying to complete.
+      @param {Object} change The data for the delta change.
+      @param {String} kind The type of delta.
+    */
+    defaultHandler: function(db, action, change, kind) {
+      console.error('Unknown delta type: ' + kind);
+      console.log(action, change);
```
Comment (mitechie) [inline app/models/handlers.js]: hmmm, we should look at getting this into our logs in the jujucharms.com future. It would be awesome to get real time feedback/errors from users once we have the back end to get that in place.

Comment (hatched) [inline app/models/handlers.js]: I have a plan for a logging/notification suite for the GUI when we have some downtime...hah!

**Comment by mitechie** [comment]
:+1: ty for the update. no QA

So this basically is setup so that the actions delta will fall through and get console logged for now. Am I reading that correct?

**Comment by makyo** [comment]
Thanks for cleaning out all of this cruft.  :+1: QAing now.

**Comment by makyo** [comment]
QA okay

**Comment by hatched** [comment]
@mitechie Correct - any non-supported deltas will log to the console and continue on. The only unsupported delta atm is the actions one.

**Comment by hatched** [comment]
Thanks for the reviews and QA!

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://localhost:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://localhost:8080/job/juju-gui-merge/856

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://localhost:8080/job/juju-gui-merge
